### PR TITLE
Add docs on how to deploy executor image to k8s/cloud

### PR DESCRIPTION
**Pull request by mrnugget**
Follow up to bug and discussion. I'd rather have something here than nothing, so I don't think this requires a lot of polish.

### CODE REVIEW

File: doc/dev/how-to/deploy_executor_image.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# How to deploy a new executor image
```
Comment (eseliger) [inline doc/dev/how-to/deploy_executor_image.md:1]:
```suggestion
# How to deploy a new executor version
```
? not sure. don't feel strongly

**Review by efritz — APPROVED** [review]

File: doc/dev/how-to/deploy_executor_image.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+* AWS credentials set as env vars: `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`
+
+## Steps
+
+1. Make a change to the `executor` code so that the buildkite build results in new `executor` AWS/GCP images being built & published. Or push to a branch with the [`executor-patch-notest/`](https://github.com/sourcegraph/sourcegraph/blob/882ed49014bc470a3be17ce74764a856f82bee4e/enterprise/dev/ci/internal/ci/runtype.go#L65-L67) prefix to trigger the build.
+1. Look at the buildkite build. Example: https://buildkite.com/sourcegraph/sourcegraph/builds/116966
+
+    There are two steps with `executor-image` in here. The first one builds and uploads image. Second step releases it.
+
+    Try to find the image name, i.e. `executor-cc28c728e5-116966`
```
Comment (eseliger) [inline doc/dev/how-to/deploy_executor_image.md:23]: That sounds like a wish 😆

File: doc/dev/how-to/deploy_executor_image.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+2. In the `infrastructure` repository, open `executors/gcp.tf` and update the image at the top in `gcp_executor_machine_image`.
+3. Create a pull request with that change.
```
Comment (eseliger) [inline doc/dev/how-to/deploy_executor_image.md:26]: The same would apply for the aws ami.

**Review by eseliger — APPROVED** [review]
Thanks for writing this down. 🚀  I think we need to adjust this a little further later, because along the process we usually would want the terraform modules to use a working version of the executors as well, but that all hasn't been specced out super well yet so this is a good start.

**Pull request merged**